src/Microsoft.PowerShell.Commands.Management/commands/management/Navigation.cs

```diff
@@ -3266,6 +3266,10 @@ protected override void ProcessRecord()
                     {
                         // not a directory
                     }
+                    catch (System.ArgumentException)
```

Comment:
Maybe it is better to skip that <DirectoryInfo code section> altogether if provider is not FileSystem provider?
e.g.: resolvedPath.Provider.Name.Equals(FileSystemProvider.ProviderName, StringComparison.OrdinalIgnoreCase)

Reply:
... that <DirectoryInfo code section> section altogether ...

Reply:
... that "DirectoryInfo code section" section altogether ...

Reply:
I agree checking for the provider seems to be a better option.

Reply:
will make the change